packages/ui/src/FileCard/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { Button } from "primereact/button";
+import { Card } from "primereact/card";
+import { ReactNode } from "react";
+
+import { IFile } from "../FilesTable";
+
+export type Messages = {
+  downloadCountLabel?: string;
+  downloadButtonLabel?: string;
+  lastDownloadedAtHeader?: string;
+  uploadedbyHeader?: string;
+  uploadedAtHeader?: string;
+  viewButtonLabel?: string;
+};
+
+type FileCardType = {
+  file: IFile;
+  messages?: Messages;
+  onArchive?: (arguments_: IFile) => void;
+  onDelete?: (arguments_: IFile) => void;
+  onDownload?: (arguments_: IFile) => void;
+  onShare?: (arguments_: IFile) => void;
+  onView?: (arguments_: IFile) => void;
+  renderThumbnail?: (arguments_: IFile) => ReactNode;
+  showThumbnail?: boolean;
+  showDescription?: boolean;
+};
+
+export const FileCard = ({
+  file,
+  onArchive,
+  onDelete,
+  onDownload,
+  onShare,
+  onView,
+  messages,
+  renderThumbnail,
+  showThumbnail = true,
+  showDescription = true,
+}: FileCardType) => {
+  const checkUploadedByData = (data: any) => {
+    if (!data.uploadedBy) {
+      return <code>&#8212;</code>;
+    }
+
+    if (data.uploadedBy.givenName || data.uploadedBy.lastName) {
+      return `${data.uploadedBy.givenName || ""} ${
+        data.uploadedBy.lastName || ""
+      }`;
+    }
+
+    return data.uploadedBy.email;
+  };
+
+  const renderIconThumbnail = () => {
+    if (showThumbnail) {
+      if (!renderThumbnail) {
+        return (
+          <div className="file-icon">
+            <i className="pi pi-file-pdf"></i>;
+          </div>
+        );
+      }
+      return <div className="file-icon">{renderThumbnail?.(file)}</div>;
+    }
+    return null;
+  };
+
+  return (
+    <Card className="file-card">
+      <div className="file-card-wrapper">
+        {renderIconThumbnail()}
+        <div className="file-details-wrapper">
+          <div className="file-details">
+            <span>
+              <span className="file-name">{file.filename}</span>
+              <span className="file-size">
+                {file.size && `(${file?.size})`}
+              </span>
+            </span>
+            {showDescription && (
+              <>
+                <span className="file-description-details">
+                  <span className="file-description">
+                    {file.description || "This is my file desription"}
+                  </span>
+                  <Button icon="pi pi-pencil" text size="small" />
+                </span>
+              </>
+            )}
```

Comment:
- No hardcoded values. file.description can be null. This part should only be displayed if description is available.
- span inside span is not a very good choice. I think we can update the wrapper span to div

```suggestion
            {showDescription && file.description &&  (
                <div className="file-description">
                  <span>{file.description}</span>
                  <Button icon="pi pi-pencil" text size="small" />
                </div>
            )}
```